### ISSUE TITLE
[clang-tidy] readability-convert-member-functions-to-static: false positive when calling overloaded member functions in lambda

### ISSUE THREAD

**Issue by Nkaspers** [llvm/llvm-project#171626]
```cpp
class A {
public:
    void foo();
    void bar();
    void bar(int a){};
};

// Warning
void A::foo() {
    auto fun = [&](auto a) {
        bar(a);
    };
}

// No Warning
void A::bar() {
    auto fun = [&](int a) {
        bar(a);
    };
}
```

The warning only shows if the member function has overloads and is called with a lambda argument of type `auto`.

**Comment by ItsGunjit** [comment]
Could you assign me this issue?
